test/integration/hds_integration_test.cc

```diff
@@ -75,56 +67,145 @@ class HdsIntegrationTest : public HttpIntegrationTest,
+    auto* http_health_check = cluster_health_check->mutable_http_health_check();
+    http_health_check->set_use_http2(false);
+    http_health_check->set_path("/healthcheck");
+
+    return server_health_check_specifier;
+  }
+
+  static constexpr uint32_t upstream_endpoints_ = 0;
 
   IntegrationStreamDecoderPtr response_;
   std::string sub_zone_{"winter"};
-  FakeHttpConnectionPtr fake_hds_connection_;
+  FakeHttpConnectionPtr hds_fake_connection_;
   FakeStreamPtr hds_stream_;
   FakeUpstream* hds_upstream_{};
+  FakeUpstream* host_upstream_{};
   FakeUpstream* service_upstream_[upstream_endpoints_]{};
   uint32_t hds_requests_{};
-  EdsHelper eds_helper_;
+  FakeHttpConnectionPtr host_fake_connection;
+  FakeStreamPtr host_stream;
 };
 
 INSTANTIATE_TEST_CASE_P(IpVersions, HdsIntegrationTest,
                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
                         TestUtility::ipTestParamsToString);
 
-// Test connectivity of Envoy and the Server
-TEST_P(HdsIntegrationTest, Simple) {
+TEST_P(HdsIntegrationTest, SingleEndpointHealthy) {
   initialize();
+
+  // Messages
   envoy::service::discovery::v2::HealthCheckRequest envoy_msg;
-  envoy::service::discovery::v2::HealthCheckRequest envoy_msg_2;
-  envoy::service::discovery::v2::HealthCheckSpecifier server_health_check_specifier;
-  server_health_check_specifier.mutable_interval()->set_nanos(500000000); // 500ms
+  envoy::service::discovery::v2::HealthCheckSpecifier server_health_check_specifier =
+      makeHealthCheckSpecifier();
 
   // Server <--> Envoy
-  fake_hds_connection_ = hds_upstream_->waitForHttpConnection(*dispatcher_);
-  hds_stream_ = fake_hds_connection_->waitForNewStream(*dispatcher_);
+  hds_fake_connection_ = hds_upstream_->waitForHttpConnection(*dispatcher_);
+  hds_stream_ = hds_fake_connection_->waitForNewStream(*dispatcher_);
   hds_stream_->waitForGrpcMessage(*dispatcher_, envoy_msg);
 
-  EXPECT_EQ(0, test_server_->counter("hds_delegate.requests")->value());
+  // Server asks for healthchecking
+  hds_stream_->startGrpcStream();
+  hds_stream_->sendGrpcMessage(server_health_check_specifier);
+
+  EXPECT_EQ(host_upstream_->httpType(), FakeHttpConnection::Type::HTTP1);
+  host_fake_connection = host_upstream_->waitForHttpConnection(*dispatcher_);
+  test_server_->waitForCounterGe("hds_delegate.requests", ++hds_requests_);
+
+  // Envoy sends a healthcheck message to an endpoint
+  host_stream = host_fake_connection->waitForNewStream(*dispatcher_);
+  host_stream->waitForEndStream(*dispatcher_);
+  EXPECT_STREQ(host_stream->headers().Path()->value().c_str(), "/healthcheck");
+  EXPECT_STREQ(host_stream->headers().Method()->value().c_str(), "GET");
+
+  // Endpoint reponds to the healthcheck
+  host_stream->encodeHeaders(Http::TestHeaderMapImpl{{":status", "200"}}, false);
+  host_stream->encodeData(1024, true);
+  test_server_->waitForCounterGe("cluster.anna.health_check.success", 1);
+
```

Comment:
We should have a wait/expect for the following `HealthCheckRequest` which contains the results of the health check from Envoy -> HDS server.

Reply:
At the moment, there is only one message sent from Envoy to the Server. The one at the start. Should we have Envoy send another message after starting the health_checker?